### PR TITLE
feat: add SourceLink to Chinook.DynamicMvvm

### PR DESCRIPTION
**Pull request by AugustinBo**
GitHub Issue: #

## Proposed Changes

- Feature

This PR includes the changes needed to support SourceLink.

## Checklist

Please check if your PR fulfills the following requirements:

- [ ] Documentation has been added/updated
- [ ] Automated Unit / Integration tests for the changes have been added/updated
- [ ] Contains **NO** breaking changes
- [ ] Updated the Changelog
- [ ] Associated with an issue

### CODE REVIEW

**Comment by AugustinBo** [comment]
SourceLink might not be necessary for the test project. It can be removed if that's the case.

File: src/DynamicMvvm.Abstractions/DynamicMvvm.Abstractions.csproj [hunk 1]
```diff
@@ -9,10 +9,19 @@
...
+		<SymbolPackageFormat>snupkg</SymbolPackageFormat>
 	</PropertyGroup>
 
 	<ItemGroup>
 		<PackageReference Include="Microsoft.Extensions.DependencyInjection.Abstractions" Version="3.1.0" />
 		<PackageReference Include="Microsoft.Extensions.Logging.Abstractions" Version="3.1.0" />
 	</ItemGroup>
+
+	<ItemGroup>
+		<PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.1.1" PrivateAssets="All"/>
```
Comment (jeanplevesque) [inline src/DynamicMvvm.Abstractions/DynamicMvvm.Abstractions.csproj:25]: Could those be in the same item group as the other package references?

**Comment by jeanplevesque** [comment]
> SourceLink might not be necessary for the test project. It can be removed if that's the case.

@AugustinBo Right, the test project is not to be packed, so it should not have SourceLink enabled.

File: src/DynamicMvvm.Tests/DynamicMvvm.Tests.csproj [hunk 2]
```diff
@@ -20,4 +25,8 @@
     <ProjectReference Include="..\DynamicMvvm.Reactive\DynamicMvvm.Reactive.csproj" />
     <ProjectReference Include="..\DynamicMvvm\DynamicMvvm.csproj" />
   </ItemGroup>
+
+  <ItemGroup>
+		<PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.1.1" PrivateAssets="All"/>
```
Comment (jeanplevesque) [inline src/DynamicMvvm.Tests/DynamicMvvm.Tests.csproj:30]: Yeah we need to undo those changes for the test project.

**Review by jeanplevesque — CHANGES_REQUESTED** [review]

File: src/DynamicMvvm.Abstractions/DynamicMvvm.Abstractions.csproj (hunk 1 above)
Comment (AugustinBo) [reply src/DynamicMvvm.Abstractions/DynamicMvvm.Abstractions.csproj:25]: I don't see any mentions that it wouldn't work.

**Review by jeanplevesque — APPROVED** [review]

**Pull request merged**